app/Console/Commands.php

```diff
@@ -10,84 +11,23 @@
  */
 class Commands
 {
-    private static function isDebugMode()
-    {
-        return config('app.debug');
-    }
-
-    public static function getCompletedPrintingJobs()
-    {
-        $command = config('commands.lpstat') . " " . config('print.stat_additional_args') . " -W completed -o " . config('print.printer_name') . " | awk '{print $1}'";
-        if (self::isDebugMode()) {
-            $result = [0];
-        } else {
-            $result = [];
-            exec($command, $result);
-        }
-        Log::info([$command, $result]);
-        return $result;
-    }
-
-    public static function print($command)
-    {
-        if (self::isDebugMode()) {
-            $job_id = 0;
-            $result = "request id is " . config('print.printer_name') . "-" . $job_id . " (1 file(s))";
-        } else {
-            $result = exec($command);
-        }
-        Log::info([$command, $result]);
-        return $result;
-    }
-
-    public static function cancelPrintJob(string $jobID)
-    {
-        $command = config('commands.cancel') . " " . $jobID;
-        if (self::isDebugMode()) {
-            // cancel(1) exits with status code 0 if it succeeds
-            $result = ['output' => '', 'exit_code' => 0];
-        } else {
-            $output = exec($command, $result, $exit_code);
-            $result = ['output' => $output, 'exit_code' => $exit_code];
-        }
-        Log::info([$command, $result]);
-        return $result;
-    }
-
-    public static function getPages($path)
-    {
-        $command = config('commands.pdfinfo') . " " . $path . " | grep '^Pages' | awk '{print $2}' 2>&1";
-        if (self::isDebugMode()) {
-            $result = rand(1, 10);
-        } else {
-            $result = exec($command);
-        }
-        Log::info([$command, $result]);
-        return $result;
-    }
-
     public static function pingRouter($router)
     {
-        if (self::isDebugMode()) {
-            $result = rand(1, 10) > 9 ? "error" : '';
-        } else {
-            // This happens too often to log.
-            $command = config('commands.ping') . " " . $router->ip . " -c 1 | grep 'error\|unreachable'";
-            $result = exec($command);
+        if (preg_match('/[a-zA-Z]/', $router->ip)) {
```

Comment:
Nitpick, as this is already an improvement: could we check if it is in fact an IP address, and if not, then throw an exception? Like using this regex? `^(?:(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])\.){3}(?:25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$`
Some quite painful shellscript could be injected here even if they do not control any letters. (Like a fork bomb: `:(){ :|:& };:`)